lib/Common/Memory/LargeHeapBlock.cpp

```diff
@@ -556,9 +615,28 @@ LargeHeapBlock::Mark(void* objectAddress, MarkContext * markContext)
         return;
     }
 
+    if (header->isPageHeapAlloc)
+    {
+        this->VerifyPageHeapPattern();
+    }
+
     DUMP_OBJECT_REFERENCE(markContext->GetRecycler(), objectAddress);
 
-    if (!UpdateAttributesOfMarkedObjects(markContext, objectAddress, header->objectSize, header->GetAttributes(this->heapInfo->recycler->Cookie),
+    unsigned char attributes = header->GetAttributes(this->heapInfo->recycler->Cookie);
+    size_t objectSize = header->objectSize;
+    if (this->InPageHeapMode())
+    {
+        // trim off the trailing part which is not a pointer
+        objectSize = objectSize - (objectSize%sizeof(void*));
+        if (objectSize == 0)
+        {
+            // finalizable object must bigger than a pointer size because of the vtable
+            Assert((attributes & (TrackBit | FinalizeBit)) == 0);
```

Comment:
removed TrackBit

---
In reply to: [60996026](https://github.com/Microsoft/ChakraCore/pull/711#discussion_r60996026) [](ancestors = 60996026)